tests/catch/unit/stream/hipStreamValue.cc

```diff
@@ -0,0 +1,495 @@
+  HIP_CHECK(hipFree(signalPtr));
+}
+
+// hipStreamWaitValue Tests
+template <bool isBlocking, typename intT, typename TEST_T>
+void initData(intT* dataPtr, int64_t* signalPtr, TEST_T tc, std::vector<hipEvent_t>& events) {
+  // Initialize memory to be waited on
+  *signalPtr = isBlocking ? tc.signalValueFail : tc.signalValuePass;
+
+
+  // Initialize host pointers
+  dataPtr[0] = DATA_INIT;
+  dataPtr[1] = DATA_INIT;
+
+
+  hipEvent_t firstWriteEvent{nullptr};
+  hipEvent_t secondWriteEvent{nullptr};
+  HIP_CHECK(hipEventCreate(&firstWriteEvent));
+  HIP_CHECK(hipEventCreate(&secondWriteEvent));
+  events.push_back(firstWriteEvent);
+  events.push_back(secondWriteEvent);
+}
+
+template <bool isBlocking, typename intT, typename TEST_T>
+void syncAndCheckData(hipStream_t stream, intT* dataPtr, int64_t* signalPtr, TEST_T tc,
+                      std::vector<hipEvent_t>& events) {
+  // Ensure first part of host memory is updated
+  HIP_CHECK(hipStreamWaitEvent(stream, events[0], 0));
+  HIP_ASSERT(dataPtr[0] == DATA_UPDATE);
+  if (isBlocking) {
+    // Ensure second part of host memory isn't updated yet
+    HIP_ASSERT(hipEventQuery(events[1]) == hipErrorNotReady);
+    HIP_ASSERT(dataPtr[1] == DATA_INIT);
+    // Update value to release stream
+    *signalPtr = tc.signalValuePass;
+  }
+
+  HIP_CHECK(hipStreamSynchronize(stream));
+  HIP_ASSERT(hipEventQuery(events[1]) == hipSuccess);
+  // Finally ensure that second part of host memory is updated
+  HIP_ASSERT(dataPtr[1] == DATA_UPDATE);
+}
+
+template <typename intT> void cleanup(hipStream_t& stream, intT* dataPtr, int64_t* signalPtr) {
+  // Cleanup
+  HIP_CHECK(hipFree(signalPtr));
+  HIP_CHECK(hipHostUnregister(dataPtr));
+  HIP_CHECK(hipStreamDestroy(stream));
+}
+
+#define DEFINE_HIP_STREAM_VALUE(TYPE, BITS, ...) hipStream##TYPE##Value##BITS(__VA_ARGS__)
+
+#define CHECK_HIP_STREAM_VALUE(TYPE, BITS, ...)                                                    \
+  HIP_CHECK(DEFINE_HIP_STREAM_VALUE(TYPE, BITS, __VA_ARGS__));
+
+template <typename intT, bool isBlocking, typename TEST_T> void testWait(TEST_T tc) {
+  if (!streamWaitValueSupported()) {
+    UNSCOPED_INFO(" hipStreamWaitValue: not supported on this device , skipping ...");
+    return;
+  }
+
+  // Initialize stream
+  hipStream_t stream{nullptr};
+  HIP_CHECK(hipStreamCreate(&stream));
+
+  // Allocate Host Memory
+  std::unique_ptr<intT> dataPtr(new intT(2));
+
+  // Register Host Memory
+  HIP_CHECK(hipHostRegister(&(dataPtr.get()[0]), sizeof(intT), 0));
+  HIP_CHECK(hipHostRegister(&(dataPtr.get()[1]), sizeof(intT), 0));
+
+  // Allocate Signal Memory
+  int64_t* signalPtr;
+  HIP_CHECK(hipExtMallocWithFlags((void**)&signalPtr, 8, hipMallocSignalMemory));
+
+  std::vector<hipEvent_t> events;
+  initData<isBlocking>(dataPtr.get(), signalPtr, tc, events);
+
+  if (std::is_same<intT, int32_t>::value) {
+    CHECK_HIP_STREAM_VALUE(Write, 32, stream, &(dataPtr.get()[0]), DATA_UPDATE, writeFlag)
```

Comment:
Request to please add test cases to verify hipStreamWaitValue64/hipStreamWaitValue32 for arrays.
Need to test memory with offset